src/utils/src/main/scala/Serializer.scala

```diff
@@ -0,0 +1,223 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import java.io.{File, InputStream, ObjectOutputStream, OutputStream}
+import com.microsoft.ml.spark.FileUtilities.using
+import org.apache.hadoop.fs.Path
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.util.MLWritable
+import org.apache.spark.ml.{Pipeline, PipelineStage}
+import org.apache.spark.sql.{DataFrame, Dataset, SaveMode, SparkSession}
+import org.json4s.jackson.JsonMethods.{compact, render}
+import org.json4s.JsonDSL._
+import scala.reflect.runtime.universe._
+import scala.language.existentials
+import scala.reflect.ClassTag
+
+abstract class Serializer[O] {
```

Comment:
This class is not tied to any Params classes so I'm not sold on ParamsSerializer. Do you have any other suggestions, or should I just add some docs?